lib/prom_ex/plugins/plug_router.ex

```diff
@@ -0,0 +1,248 @@
+      otp_app = Keyword.fetch!(opts, :otp_app)
+      metric_prefix = Keyword.get(opts, :metric_prefix, PromEx.metric_prefix(otp_app, :plug_router))
+      event_prefix = Keyword.fetch!(opts, :event_prefix)
+      set_up_telemetry_proxy(event_prefix)
+
+      [
+        http_events(metric_prefix, opts)
+      ]
+    end
+
+    defp set_up_telemetry_proxy(event_prefix) do
+      :telemetry.attach(
+        {__MODULE__, :stop},
+        event_prefix ++ [:stop],
+        &__MODULE__.handle_proxy_router_event/4,
+        %{}
+      )
+
+      :telemetry.attach(
+        {__MODULE__, :exception},
+        @default_event_prefix ++ [:exception],
+        &__MODULE__.handle_proxy_router_event/4,
+        %{}
+      )
+    end
+
+    def handle_proxy_router_event(_, measurements, meta, _) do
+      :telemetry.execute(@stop_event, measurements, meta)
+    end
+
+    defp http_events(metric_prefix, opts) do
+      http_metrics_tags = [:status, :method, :path]
+
+      routers =
+        opts
+        |> Keyword.fetch!(:routers)
+        |> MapSet.new()
+
+      ignore_routes =
+        opts
+        |> Keyword.get(:ignore_routes, [])
+        |> MapSet.new()
+
+      Event.build(
+        :plug_router_http_event_metrics,
+        [
+          # Capture request duration information
+          distribution(
+            metric_prefix ++ [:http, :request, :duration, :milliseconds],
+            event_name: @stop_event,
+            measurement: :duration,
+            description: "The time it takes for the application to process HTTP requests.",
+            reporter_options: [
+              buckets: exponential!(1, 2, 12)
+            ],
+            drop: drop_ignored(ignore_routes, routers),
+            tag_values: &get_tags(&1),
+            tags: http_metrics_tags,
+            unit: {:native, :millisecond}
+          ),
+          distribution(
+            metric_prefix ++ [:http, :response, :size, :bytes],
+            event_name: @stop_event,
+            description: "The size of the HTTP response payload.",
+            reporter_options: [
+              buckets: exponential!(1, 4, 12)
+            ],
+            measurement: &resp_body_size/2,
+            drop: drop_ignored(ignore_routes, routers),
+            tag_values: &get_tags(&1),
+            tags: http_metrics_tags,
+            unit: :byte
+          ),
+          counter(
+            metric_prefix ++ [:http, :requests, :total],
+            event_name: @stop_event,
+            description: "The number of requests that have been serviced.",
+            drop: drop_ignored(ignore_routes, routers),
+            tag_values: &get_tags(&1),
+            tags: http_metrics_tags
+          )
+        ]
+      )
+    end
+
+    defp resp_body_size(_, metadata) do
+      case metadata.conn.resp_body do
+        nil -> 0
+        _ -> :erlang.iolist_size(metadata.conn.resp_body)
+      end
+    end
+
+    defp path(conn) do
+      {path, _} = conn.private.plug_route
```

Comment:
Perhaps this answers my previous question. Guessing this is the normalized route?

Reply:
Correct

Reply:
Awesome :+1: